src/netdef.c

```diff
@@ -0,0 +1,307 @@
+
+    reset_dhcp_overrides(&netdef->dhcp4_overrides);
+    reset_dhcp_overrides(&netdef->dhcp6_overrides);
+    netdef->accept_ra = NETPLAN_RA_MODE_KERNEL;
+
+    free_garray_with_destructor(&netdef->ip4_addresses, g_free);
+    free_garray_with_destructor(&netdef->ip6_addresses, g_free);
+    free_garray_with_destructor(&netdef->address_options, free_address_options);
+
+    netdef->ip6_privacy = FALSE;
+    netdef->ip6_addr_gen_mode = NETPLAN_ADDRGEN_DEFAULT;
+    FREE_AND_NULLIFY(netdef->ip6_addr_gen_token);
+
+    FREE_AND_NULLIFY(netdef->gateway4);
+    FREE_AND_NULLIFY(netdef->gateway6);
+    free_garray_with_destructor(&netdef->ip4_nameservers, g_free);
+    free_garray_with_destructor(&netdef->ip6_nameservers, g_free);
+    free_garray_with_destructor(&netdef->search_domains, g_free);
+    free_garray_with_destructor(&netdef->routes, free_route);
+    free_garray_with_destructor(&netdef->ip_rules, free_ip_rules);
+    free_garray_with_destructor(&netdef->wireguard_peers, free_wireguard_peer);
+
+    netdef->linklocal.ipv4 = FALSE;
+    netdef->linklocal.ipv6 = TRUE;
+
+    FREE_AND_NULLIFY(netdef->bridge);
+    FREE_AND_NULLIFY(netdef->bond);
+
+    FREE_AND_NULLIFY(netdef->peer);
+
+    netdef->vlan_id = G_MAXUINT; /* 0 is a valid ID */
+    netdef->vlan_link = NULL;
+
+    FREE_AND_NULLIFY(netdef->set_mac);
+    netdef->mtubytes = 0;
+    netdef->ipv6_mtubytes = 0;
+
+    FREE_AND_NULLIFY(netdef->set_name);
+    FREE_AND_NULLIFY(netdef->match.driver);
+    FREE_AND_NULLIFY(netdef->match.mac);
+    FREE_AND_NULLIFY(netdef->match.original_name);
+    netdef->has_match = FALSE;
+    netdef->wake_on_lan = FALSE;
+    netdef->wowlan = 0;
+    netdef->emit_lldp = FALSE;
+
+    if (netdef->access_points) {
+        g_hash_table_foreach_remove(netdef->access_points, free_access_point, &backend);
+        g_hash_table_destroy(netdef->access_points);
+        netdef->access_points = NULL;
+    }
+
+    FREE_AND_NULLIFY(netdef->bond_params.mode);
+    FREE_AND_NULLIFY(netdef->bond_params.lacp_rate);
+    FREE_AND_NULLIFY(netdef->bond_params.monitor_interval);
+    FREE_AND_NULLIFY(netdef->bond_params.transmit_hash_policy);
+    FREE_AND_NULLIFY(netdef->bond_params.selection_logic);
+    FREE_AND_NULLIFY(netdef->bond_params.arp_interval);
+    free_garray_with_destructor(&netdef->bond_params.arp_ip_targets, g_free);
+    FREE_AND_NULLIFY(netdef->bond_params.arp_validate);
+    FREE_AND_NULLIFY(netdef->bond_params.arp_all_targets);
+    FREE_AND_NULLIFY(netdef->bond_params.up_delay);
+    FREE_AND_NULLIFY(netdef->bond_params.down_delay);
+    FREE_AND_NULLIFY(netdef->bond_params.fail_over_mac_policy);
+    FREE_AND_NULLIFY(netdef->bond_params.primary_reselect_policy);
+    FREE_AND_NULLIFY(netdef->bond_params.learn_interval);
+    FREE_AND_NULLIFY(netdef->bond_params.primary_slave);
+    memset(&netdef->bond_params, 0, sizeof(netdef->bond_params));
+
+    FREE_AND_NULLIFY(netdef->modem_params.apn);
+    netdef->modem_params.auto_config = FALSE;
```

Comment:
That's fine, but why don't we call `memset(..., 0, ...)` , to keep it in line with the cleanup of the other structs?